Resolved conflicts

Samuel325 opened the pull request:
Resolved merge conflicts from this branch: https://github.com/P1Gaming/SkyCore/pull/77.

mspangenberg03 committed:
Work on Jelly petting

mspangenberg03 committed:
Merge branch 'develop' into m_spangenberg-jelly-petting

mspangenberg03 committed:
Work on petting

mspangenberg03 committed:
Merge branch 'develop' into m_spangenberg-jelly-petting

mspangenberg03 committed:
Petting the jelly

mspangenberg03 committed:
Merge branch 'develop' into m_spangenberg-jelly-petting

Samuel325 committed:
Merge branch 'm_spangenberg-jelly-petting' into Samuel-Ayeni-Petting-States

Samuel325 committed:
Petting States

Added a mood states enum for each of the four states. Added a required amount of pets per state array for each of the four states. States change depending on if the player meets the required amount of pets. Added a test case for the functionality and also added some UI for the jelly's current mood.

Samuel325 committed:
Merge branch 'Samuel-Ayeni-Petting-States' into Samuel-Ayeni-Petting-States-New

Samuel325 committed:
Resolved conflicts

Resolved merge conflicts from "Samuel-Ayeni-Petting-States".

Samuel325 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Programming: Petting States](https://trello.com/c/o3p8Jy2x/991-programming-petting-states)

CaseyWalshWarder reviewed (commented):
The jellies neediness (I think _currNumOfPets in Parameters) should decrease by 1 every x seconds. That setting should be editable in the inspector.

The animator should play the animation via its state machine, rather than using the animator.Play method. Otherwise, the animations will become more difficult to manage over time. If you don't have experience with the animator window, I can help after a meeting, and you should look up a tutorial for the animator window.
![image](https://github.com/P1Gaming/SkyCore/assets/111548933/248fc5fd-9807-4bc7-a916-c4e5058aba13)

There's info in the flowchart about needing to have an empty hand, but I'm not sure whether they want the player to physically pet the jelly with the cursor or what that'd involve. We probably need clarification from design.

CaseyWalshWarder commented on Assets/Tests/PlayMode/JelliesParametersTests.cs:
```diff
@@ -65,4 +65,17 @@ public IEnumerator TestIfJellyGetsHungrierAfter10Seconds()
+
+        yield return null;
+        Assert.AreEqual(MoodStates.Slightly_Needy, parameters.MoodState);
+    }
```
We haven't been testing with unit tests when we review pull requests. Maybe we should discuss doing unit tests during a meeting.

Samuel325 committed:
Petting system update

Current number of pets can be edited in the inspector. Added a pet interval variable that decreases the current number of pets as time passes. Added functionality where petting could only be done if there was no item in the players hand. I changed animator.Play to animator.SetTrigger since I'm not sure how to use the animator state components located inside of the scripts folder. Currently, petting could only be done by clicking the "pet" button unless there is clarification from design about how petting could be done.

Samuel325 commented:
Applied some of the changes according to the flowchart.

CaseyWalshWarder committed:
Minir bug fix

Removed the trigger condition from the transition from petting to idle in the animator

CaseyWalshWarder commented:
The mood text doesn't update when it decrements the number of pets. 

The name of _petInterval is confusing because it's used for the time between when you can pet, whereas in a different script, it's the time between the number of pets decrements.

Samuel325 committed:
Variable rename and minor bug fix

Renamed pet interval variable and fixed bug where mood doesn't update on decrease.

Samuel325 commented:
Refractured some code to make it a little clearer. Also fixed a bug with the mood state not decreasing.

CaseyWalshWarder reviewed (approved):
Looks good, nice job.
